pkg/ingester/limiter.go

```diff
@@ -57,98 +53,32 @@ func NewLimiter(
 	}
 }
 
-// AssertMaxSeriesPerMetric limit has not been reached compared to the current
-// number of series in input and returns an error if so.
-func (l *Limiter) AssertMaxSeriesPerMetric(userID string, series int) error {
-	if actualLimit := l.maxSeriesPerMetric(userID); series < actualLimit {
-		return nil
-	}
-
-	return errMaxSeriesPerMetricLimitExceeded
-}
-
-// AssertMaxMetadataPerMetric limit has not been reached compared to the current
-// number of metadata per metric in input and returns an error if so.
-func (l *Limiter) AssertMaxMetadataPerMetric(userID string, metadata int) error {
-	if actualLimit := l.maxMetadataPerMetric(userID); metadata < actualLimit {
-		return nil
-	}
-
-	return errMaxMetadataPerMetricLimitExceeded
+// AssertMaxSeriesPerMetric returns true if limit has not been reached compared to the current
+// number of series in input; otherwise returns false.
+func (l *Limiter) AssertMaxSeriesPerMetric(userID string, series int) bool {
```

Comment:
Nit, I'd rename these methods to:

```suggestion
func (l *Limiter) IsWithinMaxSeriesPerMetric(userID string, series int) bool {
```

As they return bool, not err now.